### PR TITLE
feat: Tabs render tab bar

### PR DESCRIPTION
**Pull request by mushan0x0**
* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [x] Add unit tests for the feature.

### CODE REVIEW

**Comment by mushan0x0** [comment]
static DefaultTabBar 用的 rc-tabs 的 ScrollableInkTabBar，不知道有什么props，所以没更新文档。我发现用了自定义tabBar，type一些样式会不生效，因为less里用的>选择器，还不知道怎么处理。

**Comment by mushan0x0** [comment]
ant-design/ant-design#11204

File: package.json [hunk 1]
```diff
@@ -85,6 +85,7 @@
     "react-lazy-load": "^3.0.12",
     "react-lifecycles-compat": "^3.0.2",
     "react-slick": "~0.23.1",
+    "react-sticky": "^6.0.3",
```
Comment (afc163) [inline package.json]: 应该放在 devDependencies 里

File: components/tabs/TabBar.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      extraContent: tabBarExtraContent,
+      style: tabBarStyle,
+    };
+
+    if (renderTabBar) {
+      RenderTabBar = renderTabBar(props);
+    } else {
+      RenderTabBar = (
+        <ScrollableInkTabBar {...props}/>
+      );
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: 括号好像是多余的。

File: components/tabs/TabBar.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import ScrollableInkTabBar from 'rc-tabs/lib/ScrollableInkTabBar';
+import { TabsProps } from './index';
+
+export default class TabBar extends React.Component<TabsProps> {
+  render() {
+    let {
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: const

File: components/tabs/TabBar.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      tabBarStyle,
+      animated = true,
+      renderTabBar,
+      tabBarExtraContent,
+    } = this.props;
+    let inkBarAnimated = typeof animated === 'object' ? animated.inkBar : animated;
+
+    let RenderTabBar: React.ReactElement<any>;
+    const props = {
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: 这个 props 要改成更有意义的命名。

File: components/tabs/TabBar.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+export default class TabBar extends React.Component<TabsProps> {
+  render() {
+    let {
+      tabBarStyle,
+      animated = true,
+      renderTabBar,
+      tabBarExtraContent,
+    } = this.props;
+    let inkBarAnimated = typeof animated === 'object' ? animated.inkBar : animated;
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: const

File: components/tabs/demo/custom-tab-bar.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
+---
+order: 12
+title:
+  zh-CN: 自定义tabBar
+  en-US: Customized bar of tab
+---
+
+## zh-CN
+
+使用react-sticky组件实现吸顶效果。
```
Comment (afc163) [inline components/tabs/demo/custom-tab-bar.md]: 中英文直接要加空格。

File: components/tabs/index.tsx [hunk 7]
```diff
@@ -77,6 +74,13 @@ export default class Tabs extends React.Component<TabsProps, any> {
     }
   }
 
+  createNewTab = (targetKey: React.MouseEvent<HTMLElement>) => {
+    const onEdit = this.props.onEdit;
```
Comment (afc163) [inline components/tabs/index.tsx]: `const { onEdit } = this.props;`

File: components/tabs/index.zh-CN.md [hunk 8]
```diff
@@ -26,6 +26,7 @@ Ant Design 依次提供了三级选项卡，分别用于不同的场景。
 | --- | --- | --- | --- |
 | activeKey | 当前激活 tab 面板的 key | string | 无 |
 | animated | 是否使用动画切换 Tabs，在 `tabPosition=top|bottom` 时有效 | boolean \| {inkBar:boolean, tabPane:boolean} | true, 当 type="card" 时为 false |
+| renderTabBar | 替换TabBar | (props:DefaultTabBarProps) => React.ReactNode | 无 |
```
Comment (afc163) [inline components/tabs/index.zh-CN.md]: 自定义 tabBar，用于二次封装标签头

**Comment by afc163** [comment]
conflict

**Comment by afc163** [comment]
https://circleci.com/gh/ant-design/ant-design/7402?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

ci is broken

**Comment by mushan0x0** [comment]
Does this failing need to be fixed?

File: components/tabs/style/index.less [hunk 9]
```diff
@@ -344,14 +344,14 @@
...
     }
-    > .@{tab-prefix-cls}-content {
+    .@{tab-prefix-cls}-content {
       padding-left: 24px;
       border-left: @border-width-base @border-style-base @border-color-split;
     }
   }
 
   &-vertical&-right {
-    > .@{tab-prefix-cls}-bar {
```
Comment (afc163) [inline components/tabs/style/index.less]: Don't remove `>`, it is for situation of nested Tabs like:

```js
<Tabs>
  <TabPane>
    <Tabs>
      <TabPane />
      <TabPane />
    </Tabs>
  </TabPane>
  <TabPane></TabPane>
</Tabs>
```

Comment (afc163) [reply components/tabs/style/index.less]: https://github.com/ant-design/ant-design/issues/4317

File: components/tabs/demo/custom-tab-bar.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
+---
+order: 12
+title:
+  zh-CN: 自定义tabBar
```
Comment (afc163) [inline components/tabs/demo/custom-tab-bar.md]: 自定义页签头

File: components/tabs/TabBar.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+      prevIcon,
+      nextIcon,
+    };
+
+    let RenderTabBar: React.ReactElement<any>;
+
+    if (renderTabBar) {
+      RenderTabBar = renderTabBar(renderProps);
+    } else {
+      RenderTabBar = <ScrollableInkTabBar {...renderProps}/>;
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: `/>` 前应该有空格。

File: components/tabs/TabBar.tsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+      extraContent: tabBarExtraContent,
+      style: tabBarStyle,
+      prevIcon,
+      nextIcon,
+    };
+
+    let RenderTabBar: React.ReactElement<any>;
+
+    if (renderTabBar) {
+      RenderTabBar = renderTabBar(renderProps);
```
Comment (afc163) [inline components/tabs/TabBar.tsx]: Tabs.DefaultTabBar 可以作为 renderTabBar 的一个参数传进去，这样就不用放在 Tabs 上了。

**Pull request merged**